### PR TITLE
Transfer value only logs & test

### PR DESCRIPTION
**Pull request by matei-p**

### CODE REVIEW

File: arwen/contexts/validator.go [hunk 1]
```diff
@@ -47,6 +47,22 @@ func (validator *wasmValidator) verifyFunctions(instance wasmer.InstanceHandler)
 	return nil
 }
 
+var protectedFunctions = map[string]bool{
+	"internalVMErrors":  true,
+	"transferValueOnly": true}
```
Comment (sasurobert) [inline arwen/contexts/validator.go:52]: protect the following as well: 	generalSCRIdentifier = "writeLog"
	signalError          = "signalError"
	completedTxEvent     = "completedTxEvent"

File: arwen/contexts/validator.go [hunk 2]
```diff
@@ -47,6 +47,22 @@ func (validator *wasmValidator) verifyFunctions(instance wasmer.InstanceHandler)
...
+
+func (validator *wasmValidator) verifyProtectedFunctions(instance wasmer.InstanceHandler) error {
+	for functionName := range instance.GetExports() {
```
Comment (sasurobert) [inline arwen/contexts/validator.go:55]: this is not backward compatible. please use flag.

Comment (sasurobert) [reply arwen/contexts/validator.go:55]: use flagManagedCryptoAPI - that is the next one which is going to be activated.

File: arwen/contexts/validator.go (hunk 1 above)
Comment (matei-p) [reply arwen/contexts/validator.go]: done

File: arwen/contexts/validator.go (hunk 2 above)
Comment (matei-p) [reply arwen/contexts/validator.go:58]: done

File: arwen/contexts/runtime.go [hunk 3]
```diff
@@ -684,6 +684,12 @@ func (context *runtimeContext) VerifyContractCode() error {
 			logRuntime.Trace("verify contract code", "error", err)
 			return err
 		}
+	} else {
```
Comment (sasurobert) [inline arwen/contexts/runtime.go:687]: not with else - be specific. add another if separatly

Comment (matei-p) [reply arwen/contexts/runtime.go]: fixed

**Review by sasurobert — DISMISSED** [review]

File: arwen/contexts/output.go [hunk 4]
```diff
@@ -292,6 +297,13 @@ func (context *outputContext) TransferValueOnly(destination []byte, sender []byt
 	senderAcc.BalanceDelta = big.NewInt(0).Sub(senderAcc.BalanceDelta, value)
 	destAcc.BalanceDelta = big.NewInt(0).Add(destAcc.BalanceDelta, value)
 
+	context.WriteLogWithIdentifier(
+		context.host.Runtime().GetContextAddress(),
+		[][]byte{destination, sender, value.Bytes()},
```
Comment (andrei-marinica) [inline arwen/contexts/output.go:302]: Isn't it more intuitive to have the sender, then the destination?

File: arwen/contexts/output.go [hunk 5]
```diff
@@ -292,6 +297,13 @@ func (context *outputContext) TransferValueOnly(destination []byte, sender []byt
 	senderAcc.BalanceDelta = big.NewInt(0).Sub(senderAcc.BalanceDelta, value)
 	destAcc.BalanceDelta = big.NewInt(0).Add(destAcc.BalanceDelta, value)
 
+	context.WriteLogWithIdentifier(
```
Comment (sasurobert) [inline arwen/contexts/output.go:300]: write log only if value.cmp(zero) > 0

File: arwen/contexts/output.go (hunk 4 above)
Comment (matei-p) [reply arwen/contexts/output.go]: updated

File: arwen/contexts/output.go (hunk 5 above)
Comment (matei-p) [reply arwen/contexts/output.go]: fixed

**Review by andrei-marinica — APPROVED** [review]

**Pull request merged**